test/test_model.py

```diff
@@ -2450,35 +2548,39 @@ def test_local_get_status(self):
     def test_status_set_is_app_not_bool_raises(self):
         for is_app_v in [None, 1, 2.0, 'a', b'beef', object]:
             with self.assertRaises(TypeError):
-                self.backend.status_set(ops.ActiveStatus, is_app=is_app_v)
+                self.backend.status_set(ops.ActiveStatus, is_app=is_app_v)  # type: ignore
 
     def test_storage_tool_errors(self):
-        test_cases = [(
-            lambda: fake_script(self, 'storage-list', 'echo fooerror >&2 ; exit 1'),
-            lambda: self.backend.storage_list('foobar'),
-            ops.ModelError,
-            [['storage-list', 'foobar', '--format=json']],
-        ), (
-            lambda: fake_script(self, 'storage-get', 'echo fooerror >&2 ; exit 1'),
-            lambda: self.backend.storage_get('foobar', 'someattr'),
-            ops.ModelError,
-            [['storage-get', '-s', 'foobar', 'someattr', '--format=json']],
-        ), (
-            lambda: fake_script(self, 'storage-add', 'echo fooerror >&2 ; exit 1'),
-            lambda: self.backend.storage_add('foobar', count=2),
-            ops.ModelError,
-            [['storage-add', 'foobar=2']],
-        ), (
-            lambda: fake_script(self, 'storage-add', 'echo fooerror >&2 ; exit 1'),
-            lambda: self.backend.storage_add('foobar', count=object),
-            TypeError,
-            [],
-        ), (
-            lambda: fake_script(self, 'storage-add', 'echo fooerror >&2 ; exit 1'),
-            lambda: self.backend.storage_add('foobar', count=True),
-            TypeError,
-            [],
-        )]
+        test_cases: typing.List[typing.Tuple[
+            typing.Callable[[], None],
+            typing.Callable[[], None],
+            typing.Type[BaseException],
+            typing.List[typing.List[str]]]] = [(
```

Comment:
Not your issue, but IMO this block would be a whole lot simpler (and even shorter!) just written out the dumb way:

```python
fake_script(self, 'storage-list', 'echo fooerror >&2 ; exit 1')
with self.assertRaises(ModelError):
    self.backend.storage_list('foobar')
self.assertEqual(fake_script_calls(self, clear=True),
                 [['storage-list', 'foobar', '--format=json']])
```

Maybe you could make that change and it would avoid the messy type annotation too?